python_graphql_client/graphql_client.py

```diff
@@ -37,14 +38,19 @@ def execute(
         variables: dict = None,
         operation_name: str = None,
         headers: dict = {},
+        session: Session = None,
         **kwargs: Any,
     ):
         """Make synchronous request to graphQL server."""
         request_body = self.__request_body(
             query=query, variables=variables, operation_name=operation_name
         )
 
-        result = requests.post(
+        post_method = requests.post
+        if session:
```

Comment:
We should validate the type that session is a Session to prevent injection.

```suggestion
        if isinstance(session, Session):
```